src/land/UberArcanistSubmitQueueEngine.php

```diff
@@ -91,6 +98,89 @@ private function identifyRevision() {
     }
   }
 
+  /**
+   * Validate if autoland tag should be added to the revision and actual landing
+   * procedure should be stopped. Return false if tagging was skipped.
+   */
+  private function tagWithAutoland($revision) {
+    // Check if explicit flags are provided
+    if (true === $this->getTbr()) {
+      return false;
+    }
+
+    // check if we have TTY otherwise it is probably automation...
+    try {
+      phutil_console_require_tty();
+    } catch (PhutilConsoleStdinNotInteractiveException $e) {
+      return false;
+    }
+
+    // do not execute check for non uber Phabricator though general
+    // recommendation would be to checkout upstream arcanist
+    if (strpos($this->getConduit()->getHost(), "uberinternal.com")===false) {
+      return false;
+    }
+
+    // Skip if SubmitQueue is not enabled
+    $workflow = $this->getWorkflow();
+    $usesSubmitQueue = nonempty(
+      $workflow->getConfigFromAnySource('uber.land.submitqueue.enable'),
+      false
+    );
+
+    if (!$usesSubmitQueue) {
+      return false;
+    }
+
+    $taggingEnabled = nonempty(
+      $workflow->getConfigFromAnySource(
+        'uber.differential.autoland-if-building'),
+      false
+    );
+
+    if (!$taggingEnabled) {
+      return false;
+    }
+
+    // Skip if summary already has #autoland or BREAKGLASS
+    $summary = idx($revision, 'summary');
+    if ($summary !== null) {
+      $hasAutolandTag = (bool)preg_match('/\b#autoland\b/i', $summary);
+      $hasBreakglass = (bool)preg_match('/\bBREAKGLASS\b/i', $summary);
+      if ($hasAutolandTag || $hasBreakglass) {
+        $this->writeInfo(3);
```

Comment:
Maybe meant to add some message not just `3`?

Reply:
Thanks. Debugging leftovers.